### ISSUE TITLE
Add Focus groups for controls

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#9101]
### Describe the project you are working on

SpaceShip Game

### Describe the problem or limitation you are having in your project

I am writing a game where I want multiple controls to act as the same control when it comes to focus and theming.

The problem is with composite controls like these...
![image](https://github.com/godotengine/godot-proposals/assets/2453707/eee50dd0-0264-455a-a2ec-ebb730e00ad5)

I want both labels and the slider to act like it's one control.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add focus groups to controls

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Basically if a control get or loses focus then all the controls with the same focus group would also get and lose focus.
Or at least act and render like it.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

It's similar to button groups and allows easier development of composite controls.
